include/deal.II/dofs/dof_handler.h

```diff
@@ -1508,10 +1447,20 @@ DoFHandler<dim, spacedim>::n_locally_owned_dofs_per_processor() const
   if (number_cache.n_locally_owned_dofs_per_processor.empty() &&
       number_cache.n_global_dofs > 0)
     {
+      MPI_Comm comm;
+
+      const parallel::TriangulationBase<dim, spacedim> *tr =
+        (dynamic_cast<const parallel::TriangulationBase<dim, spacedim> *>(
+          &this->get_triangulation()));
+      if (tr != nullptr)
+        comm = tr->get_communicator();
+      else
+        comm = MPI_COMM_SELF;
+
       const_cast<dealii::internal::DoFHandlerImplementation::NumberCache &>(
         number_cache)
         .n_locally_owned_dofs_per_processor =
-        compute_n_locally_owned_dofs_per_processor();
+        number_cache.get_n_locally_owned_dofs_per_processor(comm);
     }
   return number_cache.n_locally_owned_dofs_per_processor;
```

Comment:
For this implementation, can't we bypass this code and simply return `Utilities::MPI::all_gather(...)`?